File: USACO_Jan2024/Cannonball.py
length, start = map(int, input().split())
coords = [list(map(int, input().split())) for _ in range(length)]

position = start - 1
currentPower = 1
targetsBroken = 0
while 0 <= position < length:
    if coords[position][0] == 1:
        if abs(currentPower) >= coords[position][1]:
            coords[position][0] = 2
            targetsBroken += 1
        position += currentPower
    elif coords[position][0] == 0:
        if currentPower > 0:
            currentPower += coords[position][1]
            currentPower *= -1
        else:
            currentPower -= coords[position][1]
            currentPower *= -1
        position += currentPower
    elif coords[position][0] == 2:
        position += currentPower

print(targetsBroken)

File: USACO_Jan2024/Cannonball.cpp
#include <bits/stdc++.h>
using namespace std;

typedef pair<int, int> pii;

int main() {
    int l, s;
    cin >> l >> s;
    vector<pii> coords(l);

    for (pii& coord : coords) {
        cin >> coord.first >> coord.second;
    }

    int position = s - 1, power = 1, a = 0;
    while (position < l && position >= 0) {
        if (coords[position].first == 1) {
            if (abs(power) >= coords[position].second) {
                coords[position].first = 2;
                a++;
            } else {
            }
            position += power;
        } else if (coords[position].first == 0) {
            if (power > 0) {
                power += coords[position].second;
                power *= -1;
            } else {
                power -= coords[position].second;
                power *= -1;
            }
            
            position += power;
            
        } else if (coords[position].first == 2) {
            position += power;
        }
    }

    cout << a << endl;
}